ERCS/erc-xxxx.md

```diff
@@ -0,0 +1,201 @@
+title: Set of Wallet Capabilities for Account Abstraction Applications
+description: A way for apps and Account Abstraction wallets to communicate more advanced parameters of Account Abstraction operations
+author: Yoav Weiss (@yoavw), Alex Forshtat (@forshtat), Dror Tirosh (@drortirosh), Shahaf Nacson (@shahafn)
+discussions-to:
+status: Draft
+type: Standards Track
+category: ERC
+created:
+requires: 5792, 7702
+---
+
+## Abstract
+
+[EIP-5792](./eip-5792) defines a baseline JSON-RPC API for a communication between wallets and dapps.
+With EIP-5792, apps and wallets can communicate about any advanced features using "capabilities" - extensions
+to the base protocol that must be defined in separate documents.
+
+This proposal defines a set of "capabilities" the wallets may want to implement in order to provide a
+comprehensive support for Account Abstraction.
+
+## Motivation
+
+## Specification
+
+### Shared configuration
+
+All actions in Account Abstraction within the context of EIP-5792 must be done on a single chain and atomically.
+This means all requests to the `wallet_sendCalls` methods MUST be done:
+
+1. With the `atomicBatch` capability enabled
+2. With the `chainId` set to the same value in all calls
+
+### Static Paymaster Configuration Capability
+
+Note that use of Paymasters managed by a "paymaster web service" is described in [ERC-7677](./eip-7677).
+
+Identifier:
+
+`staticPaymasterConfiguration`
+
+Interface:
+
+```typescript
+type StaticPaymasterConfigurationCapabilityParams = Record<
+  `0x${string}`, // Chain ID
+  {
+    paymaster: string;
+    paymasterData: string;
+    paymasterValidationGasLimit: `0x${string}`;
+    paymasterPostOpGasLimit: `0x${string}`;
+  }
+>;
+```
+
+### On-chain Query Paymaster Configuration Capability
+
+Identifier:
+
+`onChainQueryPaymasterConfiguration`
+
+Interface:
+
+```typescript
+type OnChainQueryPaymasterConfigurationCapabilityParams = Record<
+  `0x${string}`, // Chain ID
+  {
+    target: `0x${string}`, // contract to query for Paymaster configuration
+    context: `0x${string}`, // hex-encoded byte array to pass to the configuration provider
+  }
+>
+```
+
+We then define the following Solidity interface:
+
+```solidity
+
+    struct PaymasterConfiguration {
+        address paymaster;
+        bytes paymasterData;
+        uint256 paymasterValidationGasLimit;
+        uint256 paymasterPostOpGasLimit;
+    }
+
+interface IPaymasterConfigurationResolver {
+    function getPaymasterConfiguration(bytes operation, bytes context) external;
+}
+
+```
+
+The wallet MUST perform an ABI-encoding of the entire `operation` object and make a view call
+to the `getPaymasterConfiguration` function of the `target` address.
+
+If the view call fails or returns an invalid data, the `wallet_sendCalls` method must fail and return with an error.
+
+### Validity Time Range Capability
+
+Identifier:
+
+`validityTimeRange`
+
+Interface:
+
+```typescript
+type ValidityTimeRangeCapabilityParams = Record<
```

Comment:
I don't understand the time-range:
a wallet validation can return a time range, which is compared to "now". 
what is the usage scenario of this property?

Reply:
Wouldn't wallets want to allow dapps to specify the validity range? Seems like artificially limiting a feature we worked hard to add.